### PR TITLE
Vegan trait + extra clothes

### PR DESCRIPTION
**Pull request by Holli-Git**
#### Summary
Content "Vegan trait + more clothes"

#### Purpose of change

Wanted a bit more of a challenge tbh. Herbivore is neat, vegetarian is neat, but I wanted to see how a vegan survivor would go in the world. 

#### Describe the solution

Added a vegan trait. Can't eat animal products, can't wear animal products. Added some faux fur equivalents and expanded some items to make it less of a hassle. The biggest concern for me is most likely the cotton holsters, as they're a much cheaper way to get holsters. I made them unable to hold as much weight and being made from cotton makes it weaker, but I do not know if there's something else I can do.

#### Describe alternatives you've considered

Not making it? 

#### Testing

Lots. Tested that I couldn't wear items made with chitin, fur, leather, etc etc. Spawned in the clothes, wore em. Couldn't eat a food item with the "animal_product" flag. Mutating skins that are made of fur and such still work. Also first C++ so I had to do a lot of compiling.

### CODE REVIEW

**Comment by anoobindisguise** [comment]
Make the holsters from canvas? Tougher weave than cotton, with thicker threads too. Or even just use the "tactical holster" (made of nylon)

**Comment by Jarewill** [comment]
I couldn't find anything regarding scabbards in the files, you could also add a non-leather scabbard.
Nylon could be a potential material for it, I have one for my machete.
(It could also be set to spawn with machetes in shops so they aren't kept in the open.)

**Comment by Paprika-H** [comment]
> Make the holsters from canvas? Tougher weave than cotton, with thicker threads too. Or even just use the "tactical holster" (made of nylon)

Good idea! I'll get on that

**Comment by Paprika-H** [comment]
> I couldn't find anything regarding scabbards in the files, you could also add a non-leather scabbard. Nylon could be a potential material for it, I have one for my machete. (It could also be set to spawn with machetes in shops so they aren't kept in the open.)

Honestly I just didn't feel like dealing with the scabbards/baldrics right now

**Review by Maleclypse — APPROVED** [review]
I think this all makes sense.  Canvas holsters are real thing and have been pretty common at various points.  Very neat!

**Comment by Fris0uman** [comment]
Does it make sense to be vegan in a world with no industry left? Wearing that leather jacket is not supporting the leather industry anymore

**Comment by Zireael07** [comment]
Uh, what does industry have to do with being vegan??

**Comment by NetSysFire** [comment]
Yeah, theres multiple reasons for being vegan. One that is still valid in a world with no more industry is that you do not wish to steal a chickens unhatched young or cows baby feed.

**Comment by TheShadowFerret** [comment]
*Wonders if triffids are vegan*

File: data/json/mutations/mutations.json [hunk 1]
```diff
@@ -1809,6 +1809,15 @@
     "starting_trait": true,
     "valid": true
   },
+  {
+    "type": "mutation",
+    "id": "VEGAN",
+    "name": { "str": "Vegan" },
+    "points": -4,
+    "description": "You're very strictly vegan, eating animal products is out of the question, even if it means death.  Wearing animal products is also impossible.",
```
Comment (NetSysFire) [inline data/json/mutations/mutations.json:1817]:
```suggestion
    "description": "You're very strictly vegan, eating animal products is out of the question, even if it means death.  You also can't bear to wear clothing made of animal products, such as leather or fur.",
```

**Review by NetSysFire — COMMENTED** [review]
Probably too hard to add, but there are two things that can partly circumvent the trait.

- Using sinew as thread (out of scope?)
- Putting fur/wool stuff on the floor so you can still get warmth while sleeping or otherwise resting. This also does not prevent you from using a fur blanket to sleep.

**Comment by Paprika-H** [comment]
> Probably too hard to add, but there are two things that can partly circumvent the trait.
> 
> * Using sinew as thread (out of scope?)
> * Putting fur/wool stuff on the floor so you can still get warmth while sleeping or otherwise resting. This also does not prevent you from using a fur blanket to sleep.

I don't think those are possible, at least with my skillset

**Comment by NetSysFire** [comment]
The first one is pretty hard, yeah. The second one may be possible.

https://github.com/CleverRaven/Cataclysm-DDA/blob/45e6e56fb117f9e14783ee004ffa07ef79454868/src/character.cpp#L8252-L8263

**Comment by Fris0uman** [comment]
> Uh, what does industry have to do with being vegan??

You're not supporting the massive exploitation of animal suffering anymore by consuming product made of animals in the pre cataclysm, becasue that system doesn't exist anymore. But I guess this is beside the point for this PR.

**Comment by Zireael07** [comment]
@Fris0uman: That's only one of MANY reasons someone can be vegan. And that one, as you clearly mentioned yourself, no longer applies post-cataclysm

**Comment by Paprika-H** [comment]
> You're not supporting the massive exploitation of animal suffering anymore by consuming product made of animals in the pre cataclysm, becasue that system doesn't exist anymore. But I guess this is beside the point for this PR.

Some people also just don't like killing animals or are still uncomfortable with wearing leather. Like I do understand where you're coming from, but you can apply this logic to really any negative trait. Why would someone still hate grains if they're a good source of calories? Why would someone be pacifistic when you need to kill? Human behavior doesn't always follow what appears to be logical

**Comment by Zireael07** [comment]
No one is saying vegan in Cataclysm is not logical, there's a ton of reasons that don't have to do with pre-catalysm stuff

**Comment by Paprika-H** [comment]
> No one is saying vegan in Cataclysm is not logical, there's a ton of reasons that don't have to do with pre-catalysm stuff

Whoops, misread, my apolgoies

**Comment by I-am-Erk** [comment]
I love this but I think it should wait for stable.

**Comment by Paprika-H** [comment]
> I love this but I think it should wait for stable.

Alright!

**Comment by LyleSY** [comment]
This needs conflicts fixed, but other than that I think this is ready for primetime

**Comment by Fris0uman** [comment]
Can you add looks_like to the new items that don't have one please?

**Pull request closed without merging**